src/node.rs

```diff
@@ -0,0 +1,200 @@
+use crate::{Block, Consensus, Key, Message, Network, Round, State, ValidatorId};
+
+pub trait Node {
```

Comment:
We should make these traits async. That way we could use tokio to simulate time or I/O glitches in our unit tests.

Reply:
The core thing of this implementation so far is to be lightweight. If we put the traits async, we will add dependencies and possibly features.

We can alternatively merge this initial commit and then treat these improvements via issues and dedicated PRs. Thoughts?

Reply:
I'm fine with merging without that for now. But eventually, this should include some kind of tokio task or executor that drives progress and handles timeouts etc.

Reply:
Yep, we definitely should add an issue to keep track of the requirement

Reply:
#3